core/src/main/java/co/aikar/commands/CommandManager.java

```diff
@@ -68,6 +75,7 @@ public synchronized CommandOperationContext peek() {
         }
     });
     protected Map<String, RootCommand> rootCommands = new HashMap<>();
+    protected Table<Class<?>, Class<?>, Set<MethodHandle>> dependencyFields = HashBasedTable.create();
```

Comment:
Going to need to be Table<Class<?>, String, Set<Field>>
Table<Dependency Type, Dependency Key, Set of Fields>

And can we move it down to beside the existing dependencies.